contrib/pg_tde/src/smgr/pg_tde_smgr.c

```diff
@@ -29,23 +29,31 @@ typedef TDESMgrRelationData *TDESMgrRelation;
 static void CalcBlockIv(ForkNumber forknum, BlockNumber bn, const unsigned char *base_iv, unsigned char *iv);
 
 static InternalKey *
-tde_smgr_get_key(SMgrRelation reln, RelFileLocator *old_locator, bool can_create)
+tde_smgr_get_key(const RelFileLocatorBackend *smgr_rlocator)
+{
+	/* Do not try to encrypt/decrypt catalog tables */
+	if (IsCatalogRelationOid(smgr_rlocator->locator.relNumber))
+		return NULL;
+
+	return GetSMGRRelationKey(*smgr_rlocator);
+}
+
+static bool
+tde_smgr_should_encrypt(const RelFileLocatorBackend *smgr_rlocator, RelFileLocator *old_locator)
 {
 	TdeCreateEvent *event;
-	InternalKey *key;
 
-	if (IsCatalogRelationOid(reln->smgr_rlocator.locator.relNumber))
-	{
-		/* do not try to encrypt/decrypt catalog tables */
-		return NULL;
-	}
+	/* Do not try to encrypt/decrypt catalog tables */
+	if (IsCatalogRelationOid(smgr_rlocator->locator.relNumber))
+		return false;
 
-	/* see if we have a key for the relation, and return if yes */
-	key = GetSMGRRelationKey(reln->smgr_rlocator);
-	if (key != NULL)
-	{
-		return key;
-	}
+	/*
+	 * Make sure that even if a statement failed, and an event trigger end
+	 * trigger didn't fire, we don't accidentaly create encrypted files when
+	 * we don't have to. event above is a pointer, so it will reflect the
```

Comment:
comment is no longer 100% accurate, event is now below this call.

Reply:
fixed